### ISSUE TITLE
Breaking change in logical IDs and resource paths between 1.14.1 and 1.15.0

### ISSUE THREAD

**Issue by paulchambers** [aws-quickstart/cdk-eks-blueprints#1033]
### Describe the bug

When upgrading from 1.14.1 to 1.15.0 the logical IDs and aws:cdk:path of resources created by the blueprint change.

Using this stack

```
const stack = blueprints.EksBlueprint.builder()
    .account(account)
    .region(region)
    .version(KubernetesVersion.V1_29)
    .build(app, 'eks-blueprint');
```

Results in the following paths:

1.14.1:  "eks-blueprint/eks-blueprint/Nodegroupeks-blueprint-ng-ng/NodeGroupRole/Resource"
1.15.0:  "eks-blueprint/eks-blueprint/eks-blueprint/Nodegroupeks-blueprint-ng-ng/NodeGroupRole/Resource"

The extra nested blueprint name results in almost all resources being replaced. 

If resources are using static logical IDs* then this also results in the deployment failing entirely.

\* for example the karpenter addon uses an ID based on a hash of the cluster name and region. 

### Expected Behavior

Resource paths do not change between versions

### Current Behavior

Logical IDs and resource paths for 100s of resources have changed resulting in resource replacement and deployment failure

### Reproduction Steps

as above

### Additional Information/Context

I think this might have been introduced in this commit https://github.com/aws-quickstart/cdk-eks-blueprints/commit/2290edad8e380c48e43d67a1ded4cee5f81a196b

### CDK CLI Version

2.147.1 (build d3695d4)

### EKS Blueprints Version

1.15.0

### Node.js Version

v20.14.0

### Environment details (OS name and version, etc.)

Windows 11

**Comment by shapirov103** [comment]
@paulchambers an unfortunate consequence of the construct introduction [here](https://github.com/aws-quickstart/cdk-eks-blueprints/pull/1018) to facilitate reuse. Let me look into a way to keep it backwards compatible.

**Comment by shapirov103** [comment]
this has been addressed in the PR referenced above. There is another issue pending, once it is resolved I will release 1.15.1 with the fix, should be within a day after the fix.

**Comment by paulchambers** [comment]
@shapirov103 thanks for the speedy fix!

**Comment by vumdao** [comment]
This took me much time to understand why, looking forward for the release 1.15.

**Comment by shapirov103** [comment]
1.15.1 is released @paulchambers @vumdao. We can reopen this issue if you observe any further compatibility issues.

**Comment by paulchambers** [comment]
@shapirov103  1.15.1 has fixed the issue with the resource paths but i'm now seeing a different deployment failure. Looks related to #1036 so will add details there

**Comment by UlaJiang** [comment]
Hello @shapirov103 

When upgrading from 1.14.0 to 1.15.0 the logical IDs and aws:cdk:path of resources created by the blueprint change.

previous version 1.14.0
```
“aws:cdk:path”: “wallet-eks-prod-infra/prod-app-cluster/kubectllayer28/Resource”
```

now version 1.15.0
```
“aws:cdk:path”: “wallet-eks-prod-infra/prod-app-cluster/prod-app-cluster-ct/kubectllayer28/Resource”
```

After we trying to update the version to 1.151.1, the “aws:cdk:path” is still the same as what being generated in 1.15.0. Also tried to revert the version back to 1.14.0 still not help. 

We noticed that the newly generated cdk path is not only duplicate the `prod-app-cluster` but also with `-ct` surfix appened. 

the ClusterProvider we used is from [MngClusterProvider](https://aws-quickstart.github.io/cdk-eks-blueprints/cluster-providers/mng-cluster-provider/
)
```
    const clusterProvider = new blueprints.MngClusterProvider({ 
        ... ...
    });
```

Not sure what we missed. Thanks for any help ;)
